js/apps/events-page/dataServices/eventsService.js

```diff
@@ -16,14 +16,17 @@
 			let lastEventDateLocaleString;
 
 			angular.forEach(eventData, (eventItem) => {
-				let eventStartDateLocaleString = (new Date(eventItem.EventStart)).toLocaleDateString();
+				const fixedEvent = eventItem.EventStart ? eventItem : setStartDateForOnGoingEvent(eventItem, moment());
+
+				let eventStartDateLocaleString = (new Date(fixedEvent.EventStart)).toLocaleDateString();
 
 				if (lastEventDateLocaleString !== eventStartDateLocaleString) {
-					const eventDate = eventItem.EventStart || eventItem.OnGoingStartDate;
+					const eventDate = fixedEvent.EventStart || fixedEvent.OnGoingStartDate;
+					const filteredEvents = eventData.filter((thisEvent) => isEventOnDate(thisEvent, eventStartDateLocaleString));
```

Comment:
not blocking but I think this could be

```javascript
const filteredEvents = eventData.filter(thisEvent => isEventOnDate(thisEvent, eventStartDateLocaleString));
```